sdk/identity/Azure.Identity/src/WindowsVisualStudioCodeAdapter.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.IO;
+using System.Runtime.InteropServices;
+
+namespace Azure.Identity
+{
+    internal sealed class WindowsVisualStudioCodeAdapter : IVisualStudioCodeAdapter
+    {
+        private static readonly string s_userSettingsJsonPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "Code", "User", "settings.json");
+
+        public string GetUserSettingsPath() => s_userSettingsJsonPath;
+
+        public string GetCredentials(string serviceName, string accountName)
+        {
+            IntPtr credentials = IntPtr.Zero;
+            try
+            {
+                credentials = WindowsNativeMethods.CredRead($"{serviceName}/{accountName}", WindowsNativeMethods.CRED_TYPE.GENERIC);
```

Comment:
I think CredRead call should be outside the try.

Reply:
Fixed